cache.go

```diff
@@ -387,8 +385,10 @@ func (c *Cache) Fetch(key interface{}, ttl time.Duration, f FetchCallback) (valu
 			didLoad = true
 			return v, r, nil
 		}
-		if c.mode == ModeBlocking && r.Evicting() {
-			r.evictionWg.Wait()
+		if c.mode == ModeBlocking {
+			r.mu.Lock()
+			r.wg.Wait()
+			r.mu.Unlock()
```

Comment:
Seems like this may cause a race condition causing `Get` to block when it should return false.

The check-lock-check pattern for state change only works with default values when it is known that no readers of the map could have read a different value in the past. Any other state change than from default to the next state can't be relied upon.